### PR TITLE
feat(FEC-8102): toggle fullscreen on double click

### PR DESCRIPTION
**Pull request by odedhutzler**
### Description of the Changes

added logic to support differentiate double click from single click.

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/overlay-action/overlay-action.js [hunk 1]
```diff
@@ -85,7 +118,30 @@ class OverlayAction extends BaseComponent {
     if (this.props.isMobile) {
       return;
     }
-    this.togglePlayPause();
+
+    let currentTime = Date.now();
+
+    if (currentTime - this._firstClickTime < PLAY_PAUSE_BUFFER_TIME ) {
+      this.clickTimeout && clearTimeout(this.clickTimeout);
+      this._secondClickTime = currentTime;
```
Comment (OrenMe) [inline src/components/overlay-action/overlay-action.js]: remove all `_secondClickTime` uses.

File: src/components/overlay-action/overlay-action.js [hunk 2]
```diff
@@ -85,7 +118,30 @@ class OverlayAction extends BaseComponent {
     if (this.props.isMobile) {
       return;
     }
-    this.togglePlayPause();
+
+    let currentTime = Date.now();
+
+    if (currentTime - this._firstClickTime < PLAY_PAUSE_BUFFER_TIME ) {
+      this.clickTimeout && clearTimeout(this.clickTimeout);
```
Comment (OrenMe) [inline src/components/overlay-action/overlay-action.js]: Move this to a function:
```javascript
const cancelClickTimeout = () => {
   if (this.clickTimeout){ 
      clearTimeout(this.clickTimeout);
      this.clickTimeout = null;
   }
}
```

and use in both if conditions.

File: src/components/overlay-action/overlay-action.js [hunk 3]
```diff
@@ -76,7 +92,24 @@ class OverlayAction extends BaseComponent {
...
+   * toggle exit-enter fullscreen
+   *
+   * @returns {void}
+   * @memberof OverlayAction
+   */
+  toggleFullscreen(): void {
+    if (!this.player.isFullscreen()) {
+      this.logger.debug("Enter fullscreen");
+      this.player.enterFullscreen();
+
```
Comment (dan-ziv) [inline src/components/overlay-action/overlay-action.js]: remove empty line

File: src/components/overlay-action/overlay-action.js [hunk 4]
```diff
@@ -85,7 +118,30 @@ class OverlayAction extends BaseComponent {
     if (this.props.isMobile) {
       return;
     }
-    this.togglePlayPause();
+
+    let currentTime = Date.now();
```
Comment (dan-ziv) [inline src/components/overlay-action/overlay-action.js]: use const?

File: src/components/overlay-action/overlay-action.js [hunk 5]
```diff
@@ -85,7 +118,30 @@ class OverlayAction extends BaseComponent {
...
+      this.clickTimeout && clearTimeout(this.clickTimeout);
+      this.togglePlayPause();
+      this.toggleFullscreen();
+      this._secondClickTime = currentTime;
+      this._firstClickTime = 0;
+      return;
+    }
+
+    this._firstClickTime = currentTime;
+
```
Comment (dan-ziv) [inline src/components/overlay-action/overlay-action.js]: remove empty and redundant lines...

**Pull request merged**